test/jdk/java/lang/StableValue/StableValueTest.java

```diff
@@ -0,0 +1,273 @@
+    }
+
+    @Test
+    void orElse() {
+        StableValue<Integer> stable = StableValue.of();
+        assertEquals(VALUE, stable.orElse(VALUE));
+        assertNull(stable.orElse(null));
+        stable.trySet(VALUE);
+        assertEquals(VALUE, stable.orElse(VALUE2));
+    }
+
+    @Test
+    void orElseThrow() {
+        StableValue<Integer> stable = StableValue.of();
+        var e = assertThrows(NoSuchElementException.class, stable::orElseThrow);
+        assertEquals("No content set", e.getMessage());
+        stable.trySet(VALUE);
+        assertEquals(VALUE, stable.orElseThrow());
+    }
+
+    @Test
+    void isSet() {
+        isSet(VALUE);
+        isSet(null);
+   }
+
+    void isSet(Integer initial) {
+        StableValue<Integer> stable = StableValue.of();
+        assertFalse(stable.isSet());
+        stable.trySet(initial);
+        assertTrue(stable.isSet());
+    }
+
+   @Test
+   void testOrElseSetSupplier() {
+       StableTestUtil.CountingSupplier<Integer> cs = new StableTestUtil.CountingSupplier<>(() -> VALUE);
+       StableValue<Integer> stable = StableValue.of();
+       assertThrows(NullPointerException.class, () -> stable.orElseSet(null));
+       assertEquals(VALUE, stable.orElseSet(cs));
+       assertEquals(1, cs.cnt());
+       assertEquals(VALUE, stable.orElseSet(cs));
+       assertEquals(1, cs.cnt());
+   }
+
+    @Test
+    void testHashCode() {
+        StableValue<Integer> stableValue = StableValue.of();
+        // Should be Object::hashCode
+        assertEquals(System.identityHashCode(stableValue), stableValue.hashCode());
+    }
+
+    @Test
+    void testEquals() {
+        StableValue<Integer> s0 = StableValue.of();
+        assertNotEquals(null, s0);
+        StableValue<Integer> s1 = StableValue.of();
+        assertNotEquals(s0, s1); // Identity based
+        s0.setOrThrow(42);
+        s1.setOrThrow(42);
+        assertNotEquals(s0, s1);
+        assertNotEquals("a", s0);
+        StableValue<Integer> null0 = StableValue.of();
+        StableValue<Integer> null1 = StableValue.of();
+        null0.setOrThrow(null);
+        null1.setOrThrow(null);
+        assertNotEquals(null0, null1);
+    }
+
+    @Test
+    void toStringUnset() {
+        StableValue<Integer> stable = StableValue.of();
+        assertEquals(".unset", stable.toString());
+    }
+
+    @Test
+    void toStringNull() {
+        StableValue<Integer> stable = StableValue.of();
+        assertTrue(stable.trySet(null));
+        assertEquals("null", stable.toString());
+    }
+
+    @Test
+    void toStringNonNull() {
+        StableValue<Integer> stable = StableValue.of();
+        assertTrue(stable.trySet(VALUE));
+        assertEquals(Objects.toString(VALUE), stable.toString());
+    }
+
+    @Test
+    void toStringCircular() {
+        StableValue<StableValue<?>> stable = StableValue.of();
+        stable.trySet(stable);
+        String toString = stable.toString();
+        assertEquals("(this StableValue)", toString);
```

Comment:
Instead of checking the toString format, an option would be to use assertDoesNotThrow to ensure there was no StackOverflowException